### PR TITLE
refactor: InputBarButtonsView for better code readability and interaction with the class

### PR DESCRIPTION
**Pull request by agisilaos**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

With this PR we refactor InputBarButtonsView. It was a class that it was hard to work with. As we try to improve the code base slowly but surely, I took some first steps to improve the ergonomics and the code readability in the class:

- Introduced InputBarRowConstants 
- Introduced documentation for the entire class so its easy to understand how its being used
- Separated bigger methods into smaller ones 

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/InputBarButtonsView.swift [hunk 1]
```diff
@@ -20,38 +20,65 @@ import Foundation
 import UIKit
 import WireCommonComponents
 
+// MARK: - InputBarRowConstants
+
 private struct InputBarRowConstants {
```
Comment (caldrian) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/InputBarButtonsView.swift]:
```suggestion
private enum InputBarRowConstants {
```

**Review by caldrian — APPROVED** [review]
Maybe you could add a snapshot test in order to be able to see, if it behaves correctly.
Also please change the target branch to the current release branch.

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/InputBarButtonsView.swift [hunk 2]
```diff
@@ -69,7 +96,11 @@ final class InputBarButtonsView: UIView {
         addSubview(buttonInnerContainer)
 
         buttons.forEach {
-            $0.addTarget(self, action: #selector(anyButtonPressed), for: .touchUpInside)
+            let action = UIAction { [weak self] _ in
+                self?.anyButtonPressed()
+            }
+
+            $0.addAction(action, for: .touchUpInside)
```
Comment (markusfassbender) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/InputBarButtonsView.swift:103]: Nice to see the new API!

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/InputBarButtonsView.swift [hunk 3]
```diff
@@ -217,133 +228,266 @@ final class InputBarButtonsView: UIView {
...
+                        multiplier: constraintMultiplier,
+                        constant: offset
+                    )
                 )
             } else {
                 constraints.append(
-                    current.widthAnchor.constraint(equalTo: previous.widthAnchor)
+                    current.widthAnchor.constraint(
+                        equalTo: previous.widthAnchor
+                    )
```
Comment (markusfassbender) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/InputBarButtonsView.swift]: opinion: I wouldn't do the line breaks for a single argument, but as far as I know we don't have any guidelines yet.

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/InputBarButtonsView.swift [hunk 4]
```diff
@@ -217,133 +228,266 @@ final class InputBarButtonsView: UIView {
...
+
+    private func anyButtonPressed() {
         showRow(0, animated: true)
     }
 
-    @objc
-    private func ellipsisButtonPressed(_ button: UIButton!) {
+    private func ellipsisButtonPressed() {
         showRow(currentRow == 0 ? 1 : 0, animated: true)
     }
```
Comment (markusfassbender) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/InputBarButtonsView.swift]: Does it need to be an extension or can be just part of the main code with `// MARK`?

**Review by markusfassbender — APPROVED** [review]
Great, just some minor comments!

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/InputBarButtonsView.swift (hunk 2 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/InputBarButtonsView.swift:103]: 👍

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/InputBarButtonsView.swift (hunk 3 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/InputBarButtonsView.swift]: Fixed!

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/InputBarButtonsView.swift (hunk 4 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/InputBarButtonsView.swift]: Fixed it!

**Pull request merged**